teuthology/exceptions.py

```diff
@@ -50,21 +54,30 @@ class CommandFailedError(Exception):
     """
     Exception thrown on command failure
     """
-    def __init__(self, command, exitstatus, node=None, label=None):
-        self.command = command
+    def __init__(self, cmd, exitstatus, node=None, label=None):
+        self.cmd = cmd
         self.exitstatus = exitstatus
         self.node = node
         self.label = label
 
     def __str__(self):
+        if isinstance(self.cmd, (list, tuple)):
+            self.cmd = [str(Raw) if isinstance(x, Raw) else x \
+                            for x in self.cmd]
+            self.cmd = ' '.join(self.cmd)
+        elif isinstance(self.cmd, str):
+            self.cmd = sub(r'Raw\([\'"](.*)[\'"]\)', r'\1', self.cmd)
+        else:
+            raise RuntimeError('variable "command" is not str, list or tuple')
+
```

Comment:
Do not override the original command, just return the str representation computed in local variable.

Reply:
Okay, that would be better, I guess.